dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -719,6 +725,74 @@ public Limits setUserLimits(@ApiParam(hidden = true) @Auth User authUser,
+    public Set<String> getUserOrganizations(@ApiParam(hidden = true) @Auth User authUser,
+                                             @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @PathParam("gitRegistry") TokenType gitRegistry) {
+        Map<String, String> repositoryUrlToName = getGitRepositoryMap(authUser, gitRegistry);
+        if (repositoryUrlToName != null) {
+            return repositoryUrlToName.values().stream().map(repository -> repository.split("/")[0]).collect(Collectors.toSet());
+        }
+        return new HashSet<>();
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/registries/{gitRegistry}/organizations/{organization}")
+    @Operation(operationId = "getUserOrganizationRepositories")
+    @ApiOperation(nickname = "getUserOrganizationRepositories", value = "Get all of the repositories for an organization for a given git registry accessible to the logged in user.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = String.class, responseContainer = "Set")
+    public Set<String> getUserOrganizationRepositories(@ApiParam(hidden = true) @Auth User authUser,
+                                            @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @PathParam("gitRegistry") TokenType gitRegistry,
+                                            @ApiParam(value = "Git organization", required = true) @PathParam("organization") String organization) {
+        Map<String, String> repositoryUrlToName = getGitRepositoryMap(authUser, gitRegistry);
+        if (repositoryUrlToName != null) {
+            return repositoryUrlToName.values().stream().filter(repository -> repository.startsWith(organization + "/")).collect(Collectors.toSet());
+        }
+        return new HashSet<>();
+    }
+
+    /**
+     * For a given user and git registry, retrieve a map of git url to repository path
+     * @param user
+     * @param gitRegistry
+     * @return mapping of git url to repository path
+     */
+    private Map<String, String> getGitRepositoryMap(User user, TokenType gitRegistry) {
+        User foundUser = userDAO.findById(user.getId());
+
+        List<Token> scTokens = tokenDAO.findByUserId(foundUser.getId())
+                .stream()
+                .filter(token -> Objects.equals(token.getTokenSource(), gitRegistry))
+                .collect(Collectors.toList());
+
+        if (scTokens.size() > 0) {
+            Token scToken = scTokens.get(0);
+            SourceCodeRepoInterface sourceCodeRepo =  SourceCodeRepoFactory.createSourceCodeRepo(scToken, client);
+            return sourceCodeRepo.getWorkflowGitUrl2RepositoryId();
+        } else {
+            return null;
```

Comment:
As an alternative, you could have this return an empty Map in this case, then the callers don't need to do null checks.